Add `OneCycle` constructor

darsnack opened the pull request:
Closes #42 by adding a `OneCycle` convenience constructor.

### PR Checklist

- [x] Tests are added
- [x] Documentation, if applicable

darsnack committed:
Add OneCycle constructor

darsnack committed:
Add OneCycle tests

darsnack committed:
Fix typo in OneCycle constructor

isentropic commented:
anything i can do to get this one through? I kidna need the onecycle and it'd be good to make available for everyone?
also the docs are not plotting exp scehdule here https://fluxml.ai/ParameterSchedulers.jl/dev/

darsnack committed:
Make OneCycle throw BoundsError for invalid access.

- Add new Shortened schedule to cap the length of an arbitrary schedule.
- Fix BoundsError for Poly

darsnack committed:
Working implementation based on CosAnneal

darsnack committed:
Rebase for kwarg name changes

darsnack committed:
Update docs for OneCycle on main page

darsnack commented:
Nightly failures are unrelated (due to Zygote compilation).
